accepted/future-releases/small-features-21Q1/implementation-plan.md

```diff
@@ -0,0 +1,153 @@
+* [Proposal](https://github.com/dart-lang/language/blob/master/working/small-features-21q1/feature-specification.md)
+
+## Phase 0 (Prerequisite)
+
+### "generic-metadata" Experimental flag
+
+The implementation of parts of this feature ("generic metadata" and "generic function type arguments") should be developed behind an [experiment
+flag][]. Tools must be passed the flag
+`--enable-experiment=generic-metadata` to enable those features.
+
+[experiment flag]: https://github.com/dart-lang/sdk/blob/master/docs/process/experimental-flags.md
+
+While this feature is under development, individual tools may have incomplete or
+changing implementations behind the flag. When all tools have completely
+implemented the feature, the the feature will be enabled by default, and the
+flag removed in a stable release.
+
+### "triple-shift" Experimental flag
+
+The existing "triple-shift" experiment flag is already used for the partly implemented `>>>` operator. The flag is retained until we release the feature.
+
+### Tests
+
+The language team adds tests for the feature.
+
+## Phase 1 (Foundation)
+
+### CFE
+
+The CFE implements parsing the new syntax, type checking it, and compiling it to
+Kernel. Since the CFE performs constant evaluation, it also
+implements the evaluation of metadata annotations.
+
+The generic metadata feature can likely be implemented entirely in the front end and analyzer, with no back-end
+support needed. Since the front-end already evaluates the constants, and might infer type arguments, back-ends are likely used to seeing the filled-in result anyway. The analyzer deals with *source*, and must be able to make the distinction between an omitted-and-inferred type argument and an explicit type argument on a metadata constructor invocation.
+
+The generic function type argument feature can likely be *implemented* entirely in the front-end, by not disallowing generic function types as type arguments, but back-ends might need to find places where they assume that a type argument cannot be a generic function type.
+
+The `>>>` feature is already implemented in the CFE.
+
+### Analyzer
+
+For generic metadata, the analyzer needs to represent the *source*, and therefore must be able to make the distinction between an omitted-and-inferred type argument and an explicit type argument on a metadata constructor invocation. This is important for other source-manipulating tools like the formatter. The AST for metadata invocations need to have a place to store the type arguments, which isn't there now.
+
+If the analyzer assumes, and relies on the assumption, that type arguments cannot be generic function types, then it might need to fix that.
+
+The analyzer also needs to support `const Symbol(">>>")`, but otherwise seems to support the `>>>` feature (`operator>>>`  and `#>>>` works).
+
+Otherwise the analyzer is unlikely to need significant change for any of the features.
+
+## Phase 2 (Tool Implementation)
```

Comment:
We should have sections for dartdoc and IDE tooling.  

@bwilkerson @devoncarew @jcollins-g  We talked about making space for you to plan out IDE tooling work (quick fixes, etc), do you want to add some content here in a follow up, or just have it noted here (with a tracking issue filed?).

Reply:
On line 92 of this section, it says:

> There are no new quick-fixes needed.